manifests/repomd_gpg.pp

```diff
@@ -0,0 +1,64 @@
+# == Class: certs
+# Sets up the GPG key for Pulp repository metadata signing
+class certs::repomd_gpg (
+  $enable                  = $::certs::enable_repomd_gpg,
+  $regenerate              = $::certs::regenerate_repomd_gpg,
+  $dir                     = $::certs::repomd_gpg_dir,
+  $user                    = $::certs::repomd_gpg_user,
+  $group                   = $::certs::repomd_gpg_group,
+  $gpg_name                = $::certs::repomd_gpg_name,
+  $comment                 = $::certs::repomd_gpg_comment,
+  $email                   = $::certs::repomd_gpg_email,
+) {
+  if $enable {
+
+    $gpg_key_params = "Key-Type: RSA\nKey-Length:4096\nExpire-Date: 0\n"
+    $gpg_name_param = $gpg_name ? { undef => '', default => "Name-Real: ${gpg_name}\n", }
+    $gpg_comment_param = $comment ? { undef => '', default => "Name-Comment: ${comment}\n", }
+    $gpg_email_param = $email ? { undef => '', default => "Name-Email: ${email}\n", }
+    $gpg_name_params = "${gpg_name_param}${gpg_comment_param}${gpg_email_param}"
+    $gpg_params = "${gpg_key_params}${gpg_name_params}%no-protection\n%commit\n"
+    if $regenerate {
+      file { $dir:
+        ensure => absent,
+        force  => true,
+      }
+    }
+    file { $dir:
+      ensure => directory,
+      owner  => $user,
+      group  => $group,
+      mode   => '0700',
+    } ->
+    exec { 'generate repomd gpg':
+      unless   => "/usr/bin/test -e '${dir}/secring.gpg' -o -e '${dir}/private-keys-v1.d'",
+      user     => $user,
+      group    => $group,
+      provider => shell,
+      command  => "echo '${gpg_params}' | /usr/bin/gpg --homedir '${dir}' --gen-key --batch",
+    }
+
+    $pub_file = "${dir}/pub.gpg"
+    $fp_cmd_print = "/usr/bin/gpg --homedir '${dir}' --list-secret-keys --with-colons"
+    $fp_cmd_parse = "/usr/bin/awk -F: '\$1 == \"fpr\" {print \$10; exit;}'"
+    $fp_cmd = "`${fp_cmd_print} | ${fp_cmd_parse}`"
+    exec { 'extract public repomd gpg key':
+      onlyif  => "/usr/bin/test -e '${dir}'",
+      unless  => "/usr/bin/test -e '${pub_file}'",
+      user    => $user,
+      group   => $group,
+      command => "/usr/bin/gpg --homedir '${dir}' --armor --export ${fp_cmd} --output '${pub_file}'",
+    }
+
+    file { '/etc/pulp/server/plugins.conf.d/yum_distributor.json':
```

Comment:
I'd prefer if this module didn't manage pulp resources. We combine them in puppet-katello and puppet-foreman_proxy_content.